ads/smartadserver.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {loadScript} from '../src/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function smartadserver(global, data) {
+  loadScript(global, 'https://ec-ns.sascdn.com/diff/js/smart.js', () => {
+    if (!global.document.getElementById(data.tag)) {
+      global.document.write("<div id='" + data.tag + "'></div>");
```

Comment:
Use `document.createElement` and `appendChild` to `global.document.getElementById('c')` instead. The above code actually contains an XSS because `data.tag` is under attacker control.